### PR TITLE
[Fix] Minor update improvements to zca, docs for zca whitening, and improve tests.

### PR DESCRIPTION
**Pull request by ceroytres**
### Description
Related to PR #458. Fixes and updates the documentation for zca whitening and fix some of the bugs with some of the test. Replaced .view with .reshape since is safer than .view according to [this](https://discuss.pytorch.org/t/difference-between-view-reshape-and-permute/54157/2). Also adds documentation for linear transform. 

### Status
**Work in Progress**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [x] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [x] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/color/zca.py [hunk 1]
```diff
@@ -156,30 +162,37 @@ def zca_mean(inp: torch.Tensor, dim: int = 0,
...
         then it returns the inverse ZCA matrix, otherwise it returns None.
 
     Examples:
+        >>> from kornia.color import zca_mean
         >>> x = torch.tensor([[0,1],[1,0],[-1,0],[0,-1]], dtype = torch.float32)
-        >>> transform, mean = zca_mean(x) # Returns transformation matrix and data mean
+        >>> transform_matrix, mean_vector,_ = zca_mean(x) # Returns transformation matrix and data mean
+        >>> x = torch.rand(3,20,2,2)
+        >>> transform_matrix, mean_vector, inv_transform = zca_mean(x, dim = 1, return_inverse = True)
+        >>> # transform_matrix.size() equals (12,12) and the mean vector.size equal (1,12)
```
Comment (edgarriba) [inline kornia/color/zca.py:195]: just curious, are you actually testing this in some way ? we are planning to include to doctest to keep documentation prune from errors

Comment (ceroytres) [reply kornia/color/zca.py:195]: I copy paste them into the ipython repl to verify if they work.

Comment (edgarriba) [reply kornia/color/zca.py:195]: it's fine by now, we are figuring out how to properly test documentation since it can be consistant with potential changes

Comment (ceroytres) [reply kornia/color/zca.py:195]: @edgarriba Just a heads up. I made some grammar and spelling fixes, and I tried to improve the clarity of the documentation for zca whitening in PR #567. Sorry I missed these I was pretty sleepy last night, and I wanted to put something out there given the existing documentation wasn't correct in some spots in the initial PR.
